src/commands/execute.rs

```diff
@@ -1,15 +1,15 @@
 use crate::connect::Connection;
 use edgedb_tokio::server_params::PostgresAddress;
 
-use crate::analyze;
+use crate::{analyze, options};
 use crate::commands::parser::{Common, DatabaseCmd, ListCmd, DescribeCmd};
-use crate::commands::{self, Options};
+use crate::commands::{self, branching, Options};
 use crate::migrations::options::{MigrationCmd};
 use crate::migrations;
 use crate::print;
 
 
-pub async fn common(cli: &mut Connection, cmd: &Common, options: &Options)
+pub async fn common(cli: &mut Connection, cmd: &Common, options: &Options, cli_opts: &options::Options)
     -> Result<(), anyhow::Error>
```

Comment:
Needing to add `cli_opts` here suggests that we should not be passing-in these options to branching commands, since no other command is doing that.

From a quick glance over the code, it seems that we only use `options.create_connector()`. So I think we should replace `options::Options` with `commands::Options` in branching commands.

I'll try to do that in a separate PR.

Reply:
It is possible! https://github.com/edgedb/edgedb-cli/pull/1288

Reply:
Should I revert the changes to the statements trait?

Reply:
Does this change get us around the rust bug? I though it was unrelated?

I'm -0.5 (on the scale from -1 to 1) for the change of `MigrationToText`'s associated type to a generic param. It makes more sense to me for it to be an associated type. But if rust cannot handle that, generic param is fine.

Reply:
I'll verify whether your changes fixes it, my guess is it _should_ since the bug derives from passing `options` in interactive

Reply:
Seems to still be an issue, even after your changes:
```
error: implementation of `std::marker::Send` is not general enough
   --> src\interactive.rs:144:18
    |
144 |     let handle = runtime.spawn(_main(options, state, cfg));
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ implementation of `std::marker::Send` is not general enough
    |
    = note: `std::marker::Send` would have to be implemented for the type `&()`
    = note: ...but `std::marker::Send` is actually implemented for the type `&'0 ()`, for some specific lifetime `'0`
```

Reply:
Don't bother with this then, generic param is fine.